computation_graph/graph_runners.py

```diff
@@ -77,10 +78,13 @@ def unary_with_state_and_expectations(
 def variadic_with_state_and_expectations(g, sink):
     f = run.to_callable_strict(g)
 
-    def inner(turns):
+    async def inner(turns):
         prev = {}
         for turn, expectation in turns:
-            prev = f(prev, turn)
+            if inspect.iscoroutinefunction(f):
+                prev = await f(prev, turn)
+            else:
+                prev = f(prev, turn)
```

Comment:
I would use `asyncio.iscoroutinefunction` (this is what we normally use). Also I would create 1 inner function with async (similar to the previous one but with await), one inner function without (similar to what it was), and then return the correct inner function based on the is coroutine function condition. By doing this you will save a lot of time because you will do it when "building" this function and not when running it. Also can you please add a test for the async case?

Reply:
> 

Thanks! fixed